types/lib/amazon-pay.d.ts

```diff
@@ -0,0 +1,34 @@
+import { Emitter } from './emitter';
+
+export type AmazonPayOptions = {
+  /**
+   * 2 Digit Country Code
+   */
+  region: string;
+
+  /**
+   * The customer's locale. This is used to set the language rendered in the UI.
+   */
+  locale: string;
+
+  /**
+   * The currency of the payment.
+   */
+  currency: string;
+
+  /**
+   * Specify which Payment Gateway in Recurly must handle the payment.
+   */
+  gatewayCode?: string
+};
+
+export type AmazonPayEvent = 'token' | 'error' | 'close';
```

Comment:
The `AmazonPay` class directly emits a `ready` event.

The `Frame`  class will emit `error`, `done`, and `close`. So you'll need to include `done` in this too.

Reply:
Should I add `closed` instead of `close` as the `Frame` emits closed?